### PR TITLE
Added inverse functions for the ptr package

### PR DESCRIPTION
**Pull request by ellistarn**
# Changes

- :gift: Added corresponding Value functions for each ptr function. 

/kind enhancement

No issue, I just find myself needing these functions to avoid NPE errors in defaulting scenarios. This also exists within the [aws sdk](https://github.com/aws/aws-sdk-go/blob/e2d6cb448883e4f4fcc5246650f89bde349041ec/aws/convert_types.go), but I don't want to import *that* kitchen sink.

**Release Note**

```release-note
Added corresponding Value functions for each ptr function.
```

**Docs**

### CODE REVIEW

File: ptr/value_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (n3wscott) [inline ptr/value_test.go:2]: nit: 2021

File: ptr/value.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (n3wscott) [inline ptr/value.go:2]: nit: 2021

File: ptr/value_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+*/
+
+package ptr
+
+import (
+	"testing"
+	"time"
+)
+
+func TestInt32Value(t *testing.T) {
```
Comment (n3wscott) [inline ptr/value_test.go:24]: I think you need to also add the nil cases here to get better code coverage, as that is the point you are adding the functions.

**Review by n3wscott — COMMENTED** [review]
/ok-to-test

File: ptr/value_test.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+func TestFloat64Value_Nil(t *testing.T) {
+	want := float64(0)
+	gotValue := Float64Value(nil)
+	if want != gotValue {
+		t.Errorf("Float64() = &%v, wanted %v", gotValue, want)
+	}
+}
+
+
```
Comment (n3wscott) [inline ptr/value_test.go:88]: the linter does not like this extra line ^

File: ptr/value.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
```
Comment (n3wscott) [inline ptr/value.go:4]: the linter also does not like that this does not match what is in https://github.com/knative/pkg/blob/main/hack/boilerplate/boilerplate.go.txt

**Comment by ellistarn** [comment]
Hang on -- I clobbered some of my tests =/

**Comment by n3wscott** [comment]
well... come now flakes!

/test pull-knative-pkg-unit-tests

**Comment by ellistarn** [comment]
image

**Comment by ellistarn** [comment]
Thank you!

**Pull request merged**